netlogo-gui/src/main/app/InterfaceTab.scala

```diff
@@ -69,12 +77,22 @@ class InterfaceTab(workspace: GUIWorkspace,
   def getInterfacePanel = iP
 
   override def requestFocus() {
-    if(iP.isFocusable && splitPane.getDividerLocation >= maxDividerLocation) iP.requestFocus()
-    else if(commandCenter != null) commandCenter.requestFocus()
+    if(iP.isFocusable && splitPane.getDividerLocation >= maxDividerLocation) {
+      iP.requestFocus
+    }
+    else if(InterfaceTab.interfacePanelHadFocus){
+      iP.requestFocus
+    } else {
+      commandCenter.requestFocus()
+    }
   }
 
   final def handle(e: Events.SwitchedTabsEvent) {
-    commandCenterAction.setEnabled(e.newTab == this)
+    if(e.newTab != this) {
+        InterfaceTab.interfacePanelHadFocus = iP.hasFocus
```

Comment:
[`Component.hasFocus`](https://docs.oracle.com/javase/7/docs/api/java/awt/Component.html#hasFocus()) has been obsoleted by Oracle in favor of [`Component.isFocusOwner()`](https://docs.oracle.com/javase/7/docs/api/java/awt/Component.html#isFocusOwner()). Let's use `isFocusOwner` here.